code/go/internal/validator/spec.go

```diff
@@ -64,7 +65,38 @@ func (s Spec) ValidatePackage(pkg Package) ve.ValidationErrors {
 	// Semantic validations
 	errs = append(errs, s.rules(rootSpec).validate(&pkg)...)
 
-	return errs
+	// Rename unclear error messages and filter out redundant errors
+	var processedErrs ve.ValidationErrors
+	msgTransforms := []struct {
+		original string
+		new      string
+	}{
+		{"Must not validate the schema (not)", "Must not be present"},
+	}
+	redundant := []string{
+		"Must validate \"then\" as \"if\" was valid",
+		"Must validate \"else\" as \"if\" was not valid",
+	}
+	for _, e := range errs {
+		for _, msg := range msgTransforms {
+			if strings.Contains(e.Error(), msg.original) {
+				processedErrs = append(processedErrs, errors.New(strings.Replace(e.Error(), msg.original, msg.new, 1)))
+			} else if !substringInSlice(e.Error(), redundant) {
+				processedErrs = append(processedErrs, e)
+			}
```

Comment:
Nit. I tend to prefer to avoid elses and have each exception on its own if:
```suggestion
			if strings.Contains(e.Error(), msg.original) {
				processedErrs = append(processedErrs, errors.New(strings.Replace(e.Error(), msg.original, msg.new, 1)))
				continue
			}
			if substringInSlice(e.Error(), redundant) {
				continue
			}
			processedErrs = append(processedErrs, e)
```
Or maybe use a switch.
```suggestion
			switch {
			case strings.Contains(e.Error(), msg.original):
				processedErrs = append(processedErrs, errors.New(strings.Replace(e.Error(), msg.original, msg.new, 1)))
			case substringInSlice(e.Error(), redundant):
			default:
				processedErrs = append(processedErrs, e)
			}
```